include/xtensor/xtiny.hpp

```diff
@@ -0,0 +1,1999 @@
+    : m_size(0)
+    , m_data()
+    {
+    }
+
+    template <class V, class R>
+    inline
+    xtiny_impl<V, runtime_size, R>::xtiny_impl(representation_type const & begin, representation_type const & end)
+    : m_size(static_cast<size_type>(std::distance(begin, end)))
+    , m_data(begin)
+    {
+    }
+
+    template <class V, class R>
+    template <class IT, class>
+    inline
+    xtiny_impl<V, runtime_size, R>::xtiny_impl(IT begin, IT end)
+    : m_size(static_cast<size_type>(std::distance(begin, end)))
+    , m_data(const_cast<representation_type>(&*begin))
+    {
+    }
+
+    template <class V, class R>
+    inline void
+    xtiny_impl<V, runtime_size, R>::reset(representation_type const & begin, representation_type const & end)
+    {
+        m_size = static_cast<size_type>(std::distance(begin, end));
+        m_data = begin;
+    }
+
+    template <class V, class R>
+    inline void
+    xtiny_impl<V, runtime_size, R>::assign(size_type n, const value_type& v)
+    {
+        XTENSOR_ASSERT_MSG(n == size(), "xtiny_impl::assign(n, v): size mismatch.");
+        for(size_type k=0; k<size(); ++k)
+        {
+            (*this)[k] = v;
+        }
+    }
+
+    template <class V, class R>
+    template <class IT, class>
+    inline void
+    xtiny_impl<V, runtime_size, R>::assign(IT begin, IT end)
+    {
+        std::ignore = end;
+        XTENSOR_ASSERT_MSG(std::distance(begin, end) == static_cast<std::ptrdiff_t>(size()),
+            "xtiny_impl::assign(begin, end): size mismatch.");
+        for(size_type k=0; k<size(); ++k, ++begin)
+        {
+            (*this)[k] = static_cast<value_type>(*begin);
+        }
+    }
+
+    template <class V, class R>
+    inline auto
+    xtiny_impl<V, runtime_size, R>::operator[](size_type i) -> reference
+    {
+        return m_data[i];
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::operator[](size_type i) const -> const_reference
+    {
+        return m_data[i];
+    }
+
+    template <class V, class R>
+    inline auto
+    xtiny_impl<V, runtime_size, R>::data() -> pointer
+    {
+        return &m_data[0];
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::data() const -> const_pointer
+    {
+        return &m_data[0];
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::size() const -> size_type
+    {
+        return m_size;
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::max_size() const -> size_type
+    {
+        return m_size;
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::capacity() const -> size_type
+    {
+        return m_size;
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::shape() const -> index_type
```

Comment:
I think the shape function is confusing and unnecessary (at the current iteration of xtiny).

Reply:
removed